### PR TITLE
Add: Sampling support for the entities

### PR DESCRIPTION
**Pull request by keyur9**
Add: Sampling support for the entities
Refactor: Bump version to 1.6.13
Refactor: Bump version to 1.8

### CODE REVIEW

File: src/main/scala/com/moesif/filter/MoesifApiFilter.scala [hunk 1]
```diff
@@ -36,10 +40,48 @@ class MoesifApiFilter @Inject()(config: MoesifApiFilterConfig)(implicit mat: Mat
...
+  private var lastTimeAppConfigFetched: DateTime = DateTime.parse("1970-01-01", DateTimeFormat.forPattern("yyyy-MM-dd"))
+  private var isAppConfigFetched: Boolean = false
+  private var appConfigModel = new AppConfigModel
 
   private val logger = Logger.getLogger("moesif.play.filter.MoesifApiFilter")
   logger.info(s"config  is $config")
 
+  def getSampleRateToUse(userId: String, companyId: String): Int = {
+    var sampleRate = appConfigModel.getSampleRate
+    if (userId != null && appConfigModel.getUserSampleRate.containsKey(userId)) sampleRate = appConfigModel.getUserSampleRate.get(userId)
```
Comment (dkm199) [inline src/main/scala/com/moesif/filter/MoesifApiFilter.scala:54]: nitpic: Can be refactored like below. nil checks are unecessary i think
appConfigModel.getUserSampleRate.getOrElse{userId,
appConfigModel.getUserSampleRate.getOrElse(companyId, appConfigModel.getSampleRate)
}

File: src/main/scala/com/moesif/filter/MoesifApiFilter.scala [hunk 2]
```diff
@@ -36,10 +40,48 @@ class MoesifApiFilter @Inject()(config: MoesifApiFilterConfig)(implicit mat: Mat
...
+    else if (companyId != null && appConfigModel.getCompanySampleRate.containsKey(companyId)) sampleRate = appConfigModel.getCompanySampleRate.get(companyId)
+    sampleRate
+  }
+
+  def getApplicationConfig() = {
+
+    if (!isAppConfigFetched || DateTime.now(DateTimeZone.UTC).isAfter(lastTimeAppConfigFetched.plusMinutes(5))) {
```
Comment (dkm199) [inline src/main/scala/com/moesif/filter/MoesifApiFilter.scala:61]: this fcn to be synchronized.

Comment (dkm199) [reply src/main/scala/com/moesif/filter/MoesifApiFilter.scala:61]: actually I think correct way to do this, have a separate thread which runs every 5 mins, and update the appConfigModel as needed. 

Regualar threads shouldn't have to make additional HTTP call in their flow.

File: src/main/scala/com/moesif/filter/MoesifApiFilter.scala [hunk 3]
```diff
@@ -130,6 +172,9 @@ class MoesifApiFilter @Inject()(config: MoesifApiFilterConfig)(implicit mat: Mat
               val eventModel = eventModelBuilder.build()
               val eventModelMasked = advancedConfig.maskContent(eventModel)
               sendEvent(eventModelMasked)
+
+              // Calling func to fetch application configuration if needed
+              getApplicationConfig()
```
Comment (dkm199) [inline src/main/scala/com/moesif/filter/MoesifApiFilter.scala:177]: dont call here

**Review by dkm199 — CHANGES_REQUESTED** [review]
need changes ^^

**Comment by keyur9** [comment]
have updated the PR - refactored getting sample rate to use function and fetching AppConfig every 5 minutes

**Review by dkm199 — APPROVED** [review]

**Comment by dkm199** [comment]
lgtm, wanna start the publish task in devops?

**Pull request merged**